src/rp2_common/hardware_uart/uart.c

```diff
@@ -86,19 +123,88 @@ uint uart_set_baudrate(uart_inst_t *uart, uint baudrate) {
         baud_fbrd = ((baud_rate_div & 0x7f) + 1) / 2;
     }
 
-    // Load PL011's baud divisor registers
+    // Need to cleanly disable UART before touching LCR
```

Comment:
in all 3 places we do `uart_disable_before_lcr_write` we also do "was_enabled".  since cr is only a 16 bit value, maybe return a negative number for cr_save when uart wasn't enabled, and avoid the code duplication

Reply:
I've rolled up the code as suggested.  I would still like to triple check everything is ok.